src/ecrc-frontend/src/components/page/informationreview/InformationReview.test.js

```diff
@@ -50,12 +112,151 @@ describe("InformationReview Component", () => {
       saveOrg,
       saveApplicationInfo
     };
-
     const infoReview = create(
       <MemoryRouter>
         <InformationReview page={page} />
       </MemoryRouter>
     );
     expect(infoReview.toJSON()).toMatchSnapshot();
   });
+
+  test("Validate checkbox", () => {
+    prepareMocks();
+    const page = {
+      header,
+      applicant,
+      org,
+      setApplicationInfo,
+      saveApplicant,
+      saveOrg,
+      saveApplicationInfo
+    };
+
+    // Call UseEffect
+    mockUseEffect();
+
+    const infoReview = shallow(<InformationReview page={page} />);
+
+    expect(setState).toHaveBeenCalledTimes(0);
+
+    infoReview
+      .find("input")
+      .at(0)
+      .props()
+      .onClick();
+    expect(setState).toHaveBeenCalledTimes(1);
+
+    clearMocks();
+  });
+
+  test("Validate Back button", () => {
+    prepareMocks();
+    const page = {
+      header,
+      applicant,
+      org,
+      setApplicationInfo,
+      saveApplicant,
+      saveOrg,
+      saveApplicationInfo
+    };
+    const infoReview = shallow(<InformationReview page={page} />);
+
+    expect(setState).toHaveBeenCalledTimes(0);
+
+    infoReview
+      .find("Button")
+      .at(0)
+      .props()
+      .onClick();
+
+    expect(setState).toHaveBeenCalledTimes(1);
+
+    clearMocks();
+  });
+
+  test("Validate Employee relationship flow", () => {
+    prepareMocks();
+    const page = {
+      header,
+      applicant,
+      org,
+      setApplicationInfo,
+      saveApplicant,
+      saveOrg,
+      saveApplicationInfo
+    };
+
+    const infoReview = shallow(<InformationReview page={page} />);
+
+    expect(setApplicationInfo).toHaveBeenCalledTimes(0);
+    expect(saveApplicant).toHaveBeenCalledTimes(0);
+    expect(saveOrg).toHaveBeenCalledTimes(0);
+    expect(saveApplicationInfo).toHaveBeenCalledTimes(0);
+
+    infoReview
+      .find("Button")
+      .at(1)
+      .props()
+      .onClick();
+
+    clearMocks();
+  });
+
+  test("Verify Employee relationship flow", () => {
+    expect(setState).toHaveBeenCalledTimes(0);
+    expect(setApplicationInfo).toHaveBeenCalledTimes(1);
+    expect(saveApplicant).toHaveBeenCalledTimes(1);
+    expect(saveOrg).toHaveBeenCalledTimes(1);
+    expect(saveApplicationInfo).toHaveBeenCalledTimes(1);
+  });
```

Comment:
I'm a little confused about how this works. So there is a test Validate Employee relationship flow...that clicks the button, but it is the next test that performs all the expects? Should this not all be part of one test?

Reply:
Yeah, thats what I was talking about. I think the promise doesn't update the mock till the test exits. Thats why I verified it in the next test. If there is a way to update the results of the axios call or wait for the promise to resolve, I can update it.